hub/scribe/service.py

```diff
@@ -1816,6 +1830,13 @@ def _get_compactify_ops(self, height: int):
         append_deletes_hashX_history = block_hashX_history_deletes.append
         block_hashX_history_puts = []
 
+        existing_status_hashers = {
+            k: v.hasher if v else None for k, v in zip(
+                self.hashXs_by_tx,
+                self.db.prefix_db.hashX_history_hasher.multi_get([(hashX,) for hashX in self.hashXs_by_tx])
+            )
+        }
+
```

Comment:
Seems like it might be possible to reduce the number of gets if this is delayed until after:

```
            if not new_history:
                 continue
```

If it is uncommon to have empty `new_history` then never mind.

Reply:
I think it's impossible for there to not be a new history, and this is meant to allow compaction independent of advancing a block. It should be ok to delete but I'd need to test this.